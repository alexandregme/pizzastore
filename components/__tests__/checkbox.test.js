import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import Checkbox from '../checkbox';
import jest from "jest-mock";

describe("Checkbox", ()=>{
  let mountedCheckbox;
  beforeEach(()=>{
    mountedCheckbox = shallow(<Checkbox />);
  });

  it('renders without crashing', () => {
    shallow(<Checkbox />);
    expect(mountedCheckbox).toMatchSnapshot();
  });

  it('expect to have a input of the type checkbox', () => {
    let input = mountedCheckbox.find('input[type="checkbox"]');
    expect(input).toHaveLength(1);
  });

  it('expect to set the label default label', () => {
    let label = mountedCheckbox.find('label');
    expect(label).toHaveLength(1);
    expect(label.text()).toEqual('default label');
  });

  it('call a function passed to it when change', ()=>{
    const mockCallBack = jest.fn();
    const mountedInputWithCallback = shallow(<Checkbox handleChange={mockCallBack} />);
    mountedInputWithCallback.find('input').simulate('change');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});


describe("Checkbox Custom component props", ()=>{
  let mountedCheckbox;
  let props = {label: 'custom label'};

  beforeEach(()=>{
    mountedCheckbox = shallow(<Checkbox {...props} />);
  });

  it('expect to set the label custom label', () => {
    let label = mountedCheckbox.find('label');
    expect(label).toHaveLength(1);
    expect(label.text()).toEqual('custom label');
  });
});

