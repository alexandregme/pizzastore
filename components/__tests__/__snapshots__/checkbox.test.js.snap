// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="input-group-text"
      >
        <label>
          <input
            checked={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          default label
        </label>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label>
          <input
            checked={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          default label
        </label>,
        "className": "input-group-text",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />,
            "default label",
          ],
          "htmlFor": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "disabled": false,
              "id": undefined,
              "onChange": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "default label",
        ],
        "type": "label",
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="input-group-text"
        >
          <label>
            <input
              checked={false}
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
            default label
          </label>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label>
            <input
              checked={false}
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
            default label
          </label>,
          "className": "input-group-text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                disabled={false}
                onChange={[Function]}
                type="checkbox"
              />,
              "default label",
            ],
            "htmlFor": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "disabled": false,
                "id": undefined,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "default label",
          ],
          "type": "label",
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
